lib/sql.js

```diff
@@ -1508,3 +1508,251 @@ SQLConnector.prototype.getInsertedId = function(model, info) {
 SQLConnector.prototype.executeSQL = function(sql, params, options, callback) {
   throw new Error(g.f('{{executeSQL()}} must be implemented by the connector'));
 };
+
+// Refactored Discovery methods
+
+SQLConnector.prototype.querySchemas = function(options) {
+  return '';
+};
+
+/**
+ * Discover database schemas
+ *
+ // * @param {Object} options Options for discovery
+ * @param {Function} [cb] The callback function
+ */
+SQLConnector.prototype.discoverDatabaseSchemas = function(options, cb) {
+  if (!cb && typeof options === 'function') {
+    cb = options;
+    options = {};
+  }
+  options = options || {};
+  var self = this;
+  this.execute(self.querySchemas(options), cb);
+};
+
+/*!
+ * Build sql for listing tables
+ * @param options {all: for all owners, owner: for a given owner}
+ * @returns {string} The sql statement
+ */
+SQLConnector.prototype.queryTables = function(options) {
```

Comment:
So we have all the functions written below that start with `query`, shall we rename all of them as well?

- queryViews
- queryColumns
- queryExportedForeignKeys
- queryPrimaryKeys
- queryForeignKeys
- querySchemas